packages/flame/lib/src/events/game_mixins/raw_keyboard_detector.dart

```diff
@@ -0,0 +1,102 @@
+import 'package:flame/src/game/game.dart';
+import 'package:flutter/services.dart';
+
+/// This mixin allows listening to raw keyboard events, bypassing the `Focus`
+/// widget in Flutter.
+///
+/// The mixin provides two event handlers that can be overridden in your game:
+/// - [onKeyEvent] fires whenever the user presses or releases any key on a
+///   keyboard. Use this to handle "one-off" key events, such as the user
+///   pressing <Space> to jump, or <Esc> to open a menu.
+/// - [onKeysPressed] fires on every game tick as long as the user presses at
+///   least some keys on the keyboard. Use this to reliably handle repeating
+///   key events, such as user pressing arrow keys to move their character, or
+///   holding down <Ctrl> to shoot continuously.
+mixin RawKeyboardDetector on Game {
+  /// The list of keys that are currently being pressed on the keyboard (or a
+  /// keyboard-like device). The keys are listed in the order in which they
+  /// were pressed, except for the modifier keys which may be listed
+  /// out-of-order on some systems.
+  List<PhysicalKeyboardKey> physicalKeysPressed = [];
+
+  /// The set of logical keys that are currently being pressed on the keyboard.
+  /// This set corresponds to the [physicalKeysPressed] list, and can be used
+  /// to search for keys in a keyboard-layout-independent way.
+  Set<LogicalKeyboardKey> logicalKeysPressed = {};
```

Comment:
Given this issue https://github.com/flutter/flutter/issues/99330 I would not duplicate this info that is already present and accessible on RawKeyboard.

Reply:
This field is copied from the `RawKeyboard` on every key event:
```dart
logicalKeysPressed = RawKeyboard.instance.keysPressed;
```

Thus, it is always kept in sync with the information available in RawKeyboard. The reason I decided to have this copy is because in `RawKeyboard` this field is defined as follows:
```dart
Set<LogicalKeyboardKey> get keysPressed => _keysPressed.values.toSet();
```
That is, they return a copy of the internal state on every access. This is good for safety, but not so much for efficiency: if your code needs to check whether a number of certain keys are pressed, there will be a new set created on every check. Even such simple methods as `isShiftPressed`/`isControlPressed`/`isAltPressed` each create two new copies of this set.

So I thought that a good solution to this is to simply keep a local copy of the set of logical keys, and make sure it is always kept up-to-date with `RawKeyboard`.

Reply:
As for `physicalKeysPressed` -- I believe it's important to know not only which keys are currently being pressed, but also in what order they were pressed. This information is currently missing in `RawKeyboard`, they return physical keys as an orderless set, and keep it internally as a dictionary.

The reason why the order may be important is a scenario like this: suppose the user presses [Left Arrow], holds it, and then presses [Right Arrow]. Now two arrows are pressed simultaneously, and the game needs to decide how to respond to this. One possible choice is to make the player character stand still; another choice is to make the player character go right, since the right arrow was pressed last. By keeping the physical keys in a list instead of a set, we give the game an opportunity to choose which behavior to implement (my personal preference is for the latter).